### PR TITLE
Add/Upgrade `weekStart` preference for Teams and Organizations

### PR DESCRIPTION
**Pull request by Cleymax**
* Add validation, default for Organization weekStart preference
* Add weekStart preference for Team

Valid days are the same as those in the web UI.

Reference: 
- https://grafana.com/docs/grafana/latest/developers/http_api/preferences/#get-current-user-prefs

### CODE REVIEW

File: internal/resources/grafana/resource_organization_preferences.go [hunk 1]
```diff
@@ -62,11 +62,12 @@ func ResourceOrganizationPreferences() *schema.Resource {
...
 			},
-			// TODO: add validation?
 			"week_start": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "The Organization week start.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The Organization week start day. Available values are `sunday`, `monday`, `saturday`, or an empty string for the default.",
+				ValidateFunc: validation.StringInSlice([]string{"sunday", "monday", "saturday", ""}, false),
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_organization_preferences.go:69]:
```suggestion
				ValidateFunc: validation.StringInSlice([]string{"sunday", "monday", "saturday", ""}, true),
```

Should it be case-insensitive?

```
=== CONT  TestAccResourceOrganizationPreferences_WithDashboardID
    resource_organization_preferences_test.go:20: Step 1/5 error: Error running pre-apply refresh: exit status 1
        
        Error: expected week_start to be one of ["sunday" "monday" "saturday" ""], got Monday
        
          with grafana_organization_preferences.test,
          on terraform_plugin_test.tf line 18, in resource "grafana_organization_preferences" "test":
          18:   week_start = "Monday"
```

Comment (Cleymax) [reply internal/resources/grafana/resource_organization_preferences.go:69]: Compared with Grafana's source code, the name of the day is fully case-sensitive no ? 

https://github.com/grafana/grafana-openapi-client-go/blob/main/models/preferences.go#L40
https://github.com/grafana/grafana/blob/main/pkg/kinds/dashboard/dashboard_spec_gen.go#L786

**Comment by julienduchesne** [comment]
Before this can be merged, the tests will have to be fixed: 
```
=== CONT  TestAccResourceOrganizationPreferences_WithDashboardID
    resource_organization_preferences_test.go:20: Step 1/5 error: Error running pre-apply refresh: exit status 1
        
        Error: expected week_start to be one of ["sunday" "monday" "saturday" ""], got Monday
        
          with grafana_organization_preferences.test,
          on terraform_plugin_test.tf line 18, in resource "grafana_organization_preferences" "test":
          18:   week_start = "Monday"
```

File: internal/resources/grafana/resource_organization_preferences.go (hunk 1 above)
Comment (Cleymax) [reply internal/resources/grafana/resource_organization_preferences.go:69]: @julienduchesne  ?

Comment (julienduchesne) [reply internal/resources/grafana/resource_organization_preferences.go:69]: You are right that the lowercase value is right and uppercase isn't. I tried `Monday` and it just doesn't work. `monday` does. We do have some invalid examples and tests that need to be changed though. Ex: https://github.com/grafana/terraform-provider-grafana/blob/master/examples/resources/grafana_organization_preferences/resource.tf#L4

Comment (Cleymax) [reply internal/resources/grafana/resource_organization_preferences.go:69]: I will modify the incorrect tests.

File: examples/resources/grafana_organization_preferences/resource.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 resource "grafana_organization_preferences" "test" {
   theme      = "light"
   timezone   = "utc"
-  week_start = "Tuesday"
+  week_start = "tuesday"
```
Comment (julienduchesne) [inline examples/resources/grafana_organization_preferences/resource.tf]: tuesday is an invalid value. We should change that. It has always been wrong

Comment (Cleymax) [reply examples/resources/grafana_organization_preferences/resource.tf]: Of course. The only values accepted are:  `sunday`, `monday`, `saturday`

File: internal/resources/grafana/resource_organization_preferences_test.go [hunk 3]
```diff
@@ -32,17 +32,17 @@ func testAccResourceOrganizationPreferences(t *testing.T, withUID bool) {
 	prefs := models.Preferences{
 		Theme:     "light",
 		Timezone:  "utc",
-		WeekStart: "Monday",
+		WeekStart: "monday",
 	}
 	updatedPrefs := models.Preferences{
 		Theme:     "dark",
 		Timezone:  "utc",
-		WeekStart: "Tuesday",
```
Comment (Cleymax) [inline internal/resources/grafana/resource_organization_preferences_test.go:40]: I also changed the day in the go tests

**Review by julienduchesne — APPROVED** [review]
Looks good! Thanks

**Pull request merged**